pyccel/ast/core.py

```diff
@@ -2065,6 +2066,41 @@ def _ignore(cls, c):
         """
         return c is None or isinstance(c, (FunctionDef, *cls._ignored_types))
 
+class IndexedFunctionCall(FunctionCall):
+    """
+    Represents an indexed function call in the code.
+
+    Class representing indexed function calls, encapsulating all
+    relevant information for such calls within the code base.
+
+    Parameters
+    ----------
+    func : FunctionDef
+        The function being called.
+
+    args : list of FunctionCallArgument
+        The arguments passed to the function.
+
+    indexes : list of TypedAstNode
+        The indexes of the function call.
```

Comment:
Much better but I would like to suggest a small improvement
```suggestion
    args : iterable of FunctionCallArgument
        The arguments passed to the function.

    indexes : iterable of TypedAstNode
        The indexes of the function call.
```

An iterable is any object over which we can iterate. This means that the constructor is expected to work for both lists and tuples

Reply:
[84c1b53](https://github.com/pyccel/pyccel-cuda/pull/42/commits/84c1b535cf9ce3a5c5cfc204c2d9da5abdd98843)